src/control/system/membership.go

```diff
@@ -198,11 +208,93 @@ func (m *Membership) Join(req *JoinRequest) (resp *JoinResponse, err error) {
 	if !IsMemberNotFound(err) {
 		return nil, err
 	}
+	err = nil
+
+	// If in replace mode, attempt to update UUID of existing member if identical control
+	// address and fabric URIs. Neither UUID or rank in request will match MS-db member.
+	if req.ReplaceMode {
+		if !req.Rank.Equals(NilRank) {
+			return nil, errors.New("unexpected rank in rejoin request")
+		}
+
+		oldCount, err := m.Count()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get member count")
+		}
+		currentMembers, err := m.db.AllMembers()
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get all members")
+		}
+
+		var memberToReplace *Member
+		for _, cm := range currentMembers {
+			// Only match identical member with different UUID.
+			switch {
+			case cm.Addr.String() != req.ControlAddr.String(),
+				cm.PrimaryFabricURI != req.PrimaryFabricURI,
+				!slices.Equal(cm.SecondaryFabricURIs,
```

Comment:
From my understanding `SecondaryFabricURIs` and `SecondaryFabricContexts` are not sorted.  Maybe we should sort them before testing checking if they have the same content.

Reply:
I was under the impression that these lists should be identical for a rejoining rank, @kjacque what do you think?

Reply:
We identify the secondary providers by an index, so the order should be consistent.